src/framework_detectors.rs

```diff
@@ -0,0 +1,173 @@
+use regex::Regex;
+
+use crate::testrun::Framework;
+
+fn gen_reg(s: &str) -> Regex {
+    Regex::new(format!(r"(?i){}(\W|$)", s).as_str()).unwrap()
+}
+
+fn apply_reg(rl: &[(Regex, Framework)], v: Vec<String>) -> Option<Framework> {
+    for (r, f) in rl {
+        for s in v.iter() {
+            if r.is_match(s) {
+                return Some(*f);
+            }
+        }
+    }
+    None
+}
+
+fn get_framework_names() -> [(Regex, Framework); 4] {
+    [
+        (gen_reg("pytest"), Framework::Pytest),
+        (gen_reg("jest"), Framework::Jest),
+        (gen_reg("vitest"), Framework::Vitest),
+        (gen_reg("phpunit"), Framework::PHPUnit),
+    ]
+}
+
+fn get_file_extensions() -> [(Regex, Framework); 2] {
+    [
+        (gen_reg(".py"), Framework::Pytest),
+        (gen_reg(".php"), Framework::PHPUnit),
+    ]
+}
+
+// i want it to iterate through running certain regexes on all
+pub fn detect_framework(
+    testsuites_name: String,
+    mut testsuite_names: Vec<String>,
+    mut filenames: Vec<String>,
+    example_class_name: String,
+    example_test_name: String,
+    failure_messages: Vec<String>,
```

Comment:
its a bit weird to take all these as separate *owned* parameters, in particular if you are mutating the `Vec`s.

If you insert / push things into those lists, it might make sense to either delegate that responsibility to the caller.
Otherwise, using `&str` and `&[String]` or `&[&str]` is preferable here.